### PR TITLE
Pre search filter

### PR DESCRIPTION
**Pull request by sarahet**
Introduces a pre-search filter for the nucleotide mode that checks whether seeds only contain a single letter.

### CODE REVIEW

File: src/search_algo.hpp [hunk 1]
```diff
@@ -529,6 +529,23 @@ search(LocalDataHolder<TGlobalHolder> & lH)
...
+            {
+                bool all_equal = true;
+                for (size_t j = seedBegin + 1; j < (seedBegin + lH.options.seedLength); j++)
+                {
+                    if (lH.redQrySeqs[i][j] != lH.redQrySeqs[i][seedBegin])
+                    {
+                        all_equal = false;
+                        break;
+                    }
+                }
```
Comment (h-2) [inline src/search_algo.hpp:543]: This can be made a bit more readable with std::equal() or std::adjacent_find, but that's 💅🏻

**Review by h-2 — COMMENTED** [review]
In general, would it maybe be a simpler (and also faster?) solution to check if the current letter equals the next and if yes `++seedBegin`? 

It could be integrated this with the ` // skip proteine 'X' or Dna 'N'` ...

What do you think?

Also, I think we can have this for protein search, too.

**Comment by sarahet** [comment]
The tests fail because of something related to the dna_n_to_random view (very weird).

Regarding your suggestion: But wouldn't we want to go forward with a seed that e.g. looks like AAGCTAGCA and not move the seedBegin in this case?

**Comment by h-2** [comment]
> The tests fail because of something related to the dna_n_to_random view (very weird).

Can you get it fixed, or shall I have a look?

> Regarding your suggestion: But wouldn't we want to go forward with a seed that e.g. looks like AAGCTAGCA and not move the seedBegin in this case?

Why not? I assume that fewer repetitions always increase the usefulness of a seed. Although, this approach will only prevent AAAAAAAT and not TAAAAAAAA.

The only situation where it would be detrimental is, if it is the last possible seed in a sequence, and we don't search at all, because we extend. I am not sure how big this effect would be, though.

**Comment by sarahet** [comment]
New solution implemented

**Pull request merged**